source/Halibut.Tests/Transport/Observability/ConnectionObserverFixture.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut.Tests.Support;
+using Halibut.Tests.Support.TestAttributes;
+using Halibut.Tests.Support.TestCases;
+using Halibut.Tests.TestServices.Async;
+using Halibut.TestUtils.Contracts;
+using NUnit.Framework;
+
+namespace Halibut.Tests.Transport.Observability
+{
+    public class ConnectionObserverFixture : BaseTest
```

Comment:
nit: would tests disposing of the Service be good to add e.g. tentacle shutdown and that causes the observer to detect a close

Reply:
I had a stab at what I thought you meant by this [here](https://github.com/OctopusDeploy/Halibut/pull/493)

The cheapest way was to just verify the closure outside the disposal of the clientAndService.

Let me know if this was not what you had in mind 🙂